### PR TITLE
feat(histogram): add meaningful exception when histograms add buckets with different sizes

### PR DESCRIPTION
**Pull request by Boyuan-Yu**
Pull Request checklist

- [x]  The commit(s) message(s) follows the contribution [guidelines](https://github.com/filodb/FiloDB/pull/CONTRIBUTING.md) ?
- [x]  Tests for the changes have been added (for bug fixes / features) ?
- [x]  Docs have been added / updated (for bug fixes / features) ?

Current behavior : when histograms add buckets with different sizes, it returns a 5xx error

New behavior : it returns an IllegalArgumentException with a meaningful message

### CODE REVIEW

**Comment by sandeep6189** [comment]
Please follow the [guidelines](https://github.com/filodb/FiloDB/blob/develop/CONTRIBUTING.md) for PR descriptions ( look at the previous closed PRs for reference as well)

**Comment by Boyuan-Yu** [comment]
> Please follow the [guidelines](https://github.com/filodb/FiloDB/blob/develop/CONTRIBUTING.md) for PR descriptions ( look at the previous closed PRs for reference as well)

Thanks for the reminder. It's updated.

File: memory/src/main/scala/filodb.memory/format/vectors/Histogram.scala [hunk 1]
```diff
@@ -154,7 +154,11 @@ final case class LongHistogram(buckets: HistogramBuckets, values: Array[Long]) e
    * Adds the buckets from other into this LongHistogram
    */
   final def add(other: LongHistogram): Unit = {
-    assert(other.buckets == buckets)
+    if (other.buckets != buckets) {
+      throw new IllegalArgumentException(
+        "Mismatch in bucket sizes. Cannot add histograms with different bucket configurations."
```
Comment (yu-shipit) [inline memory/src/main/scala/filodb.memory/format/vectors/Histogram.scala:159]:
```suggestion
           s"Mismatch in bucket sizes. Cannot add histograms with different bucket configurations. Expected: ${buckets}, Found: ${other.buckets}"
```

**Review by yu-shipit — APPROVED** [review]
lgtm

**Pull request merged**